test/sd59x18/log2/log2.t.sol

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.13;
+
+import {
+    E,
+    MAX_WHOLE_SD59x18,
+    MAX_SD59x18,
+    PI,
+    PRBMathSD59x18__LogInputTooSmall,
+    SD59x18,
+    ZERO,
+    log2
+} from "~/SD59x18.sol";
+import { SD59x18__BaseTest } from "../SD59x18BaseTest.t.sol";
+
+contract SD59x18__Log2Test is SD59x18__BaseTest {
+    function testLog2__Zero() external {
+        SD59x18 x = ZERO;
+        vm.expectRevert(abi.encodeWithSelector(PRBMathSD59x18__LogInputTooSmall.selector, x));
+        log2(x);
+    }
+
+    modifier NotZero() {
+        _;
+    }
+
+    function testLog2__Negative() external NotZero {
+        SD59x18 x = sd(-1);
+        vm.expectRevert(abi.encodeWithSelector(PRBMathSD59x18__LogInputTooSmall.selector, x));
+        log2(x);
+    }
+
+    modifier Positive() {
+        _;
+    }
+
+    function powerSets() internal returns (Set[] memory) {
+        delete sets;
+        sets.push(set({ x: 0.0625e18, expected: -4e18 }));
+        sets.push(set({ x: 0.125e18, expected: -3e18 }));
+        sets.push(set({ x: 0.25e18, expected: -2e18 }));
+        sets.push(set({ x: 0.5e18, expected: -1e18 }));
+        sets.push(set({ x: 1e18, expected: ZERO }));
+        sets.push(set({ x: 2e18, expected: 1e18 }));
+        sets.push(set({ x: 4e18, expected: 2e18 }));
+        sets.push(set({ x: 8e18, expected: 3e18 }));
+        sets.push(set({ x: 16e18, expected: 4e18 }));
+        sets.push(set({ x: 195e18, expected: 7_607330313749610674 }));
```

Comment:
This was a mistake in my JavaScript tests. 195 is not a power of two. This should have been `MAX_SD59x18.add(sd(1)).div(sd(2))` (the largest power of two which fits within `SD59x18`).